modules/navier_stokes/src/bcs/INSFEFluidEnergyDirichletBC.C

```diff
@@ -40,10 +43,11 @@ INSFEFluidEnergyDirichletBC::INSFEFluidEnergyDirichletBC(const InputParameters &
     _u_vel(coupledValueOld("u")),
     _v_vel(_mesh.dimension() >= 2 ? coupledValueOld("v") : _zero),
     _w_vel(_mesh.dimension() == 3 ? coupledValueOld("w") : _zero),
+    _T_scale(getParam<Real>("T_scale")),
+    _T_scalar(isParamValid("T_scalar") ? coupledScalarValue("T_scalar") : _zero),
     _has_vbc(parameters.isParamValid("v_fn")),
-    _has_Tbc(parameters.isParamValid("T_fn")),
     _velocity_fn(_has_vbc ? &getFunction("v_fn") : NULL),
-    _temperature_fn(_has_Tbc ? &getFunction("T_fn") : NULL)
+    _temperature_fn(getFunction("T_fn"))
 {
```

Comment:
Disallow T_fn and T_scalar at the same time?
we cant have two temperatures, would give T**2